### ISSUE TITLE
Network requests fail on Java 9 because the default storetype changed

### ISSUE THREAD

**Issue by hborders** [robolectric/robolectric#5496]
### Description
Robolectric test that uses OkHttp for real HTTP requests throws java.lang.NullPointerException: No password supplied for PKCS#12 KeyStore

### Steps to Reproduce
Make an HTTP call with `OkHttpClient` within a test run with `org.robolectric.RobolectricTestRunner`.

### Robolectric & Android Version
`testImplementation "org.robolectric:robolectric:4.3.1"`
```
    compileSdkVersion 29
    defaultConfig {
        minSdkVersion 28
        targetSdkVersion 29
        versionCode 1
        versionName "1.0"
    }
```

### Summarizing [my stackoverflow answer](https://stackoverflow.com/a/60472730/9636):
After I upgraded to the Android 10 SDK, I had to update my unit test JVM to JDK 9, my integration tests fail with:
```
java.lang.NullPointerException: No password supplied for PKCS#12 KeyStore.

    at org.bouncycastle.jcajce.provider.keystore.pkcs12.PKCS12KeyStoreSpi.engineLoad(Unknown Source)
    at java.base/java.security.KeyStore.load(KeyStore.java:1479)
    at java.base/sun.security.ssl.TrustStoreManager$TrustAnchorManager.loadKeyStore(TrustStoreManager.java:367)
    at java.base/sun.security.ssl.TrustStoreManager$TrustAnchorManager.getTrustedCerts(TrustStoreManager.java:315)
    at java.base/sun.security.ssl.TrustStoreManager.getTrustedCerts(TrustStoreManager.java:59)
    at java.base/sun.security.ssl.TrustManagerFactoryImpl.engineInit(TrustManagerFactoryImpl.java:51)
    at java.base/javax.net.ssl.TrustManagerFactory.init(TrustManagerFactory.java:278)
    at okhttp3.internal.Util.platformTrustManager(Util.java:640)
    at okhttp3.OkHttpClient.<init>(OkHttpClient.java:228)
    at okhttp3.OkHttpClient.<init>(OkHttpClient.java:202)
```

According to [JEP-229](http://openjdk.java.net/jeps/229):

 > This feature changes the default keystore type from JKS to PKCS12. By default, new keystores will be created in the PKCS12 keystore format. Existing keystores will not change and keystore applications can continue to explicitly specify the keystore type they require.
> 
> Existing applications must not be disrupted. Keystores tend to be long-lived, so we need to support access across several JDK releases. Applications that access keystores created by earlier JDK releases must run unaltered on JDK 9. Similarly, applications that access keystores created by JDK 9 should run unaltered on earlier JDK releases.
> 
> This requirement is achieved by introducing a keystore detection mechanism that understands both the JKS and PKCS12 formats. A keystore's format is examined before it is loaded to determine its type and then the appropriate keystore implementation is used to access it. The mechanism is enabled by default but can be disabled if required.
> 
> Support for this keystore-detection mechanism may be backported to earlier JDK releases.

Thus, the classic `$JAVA_HOME/lib/security/cacerts` is still a Java Key Store, which I could verify:

```
$ file /Library/Java/JavaVirtualMachines/jdk-9.0.4.jdk/Contents/Home/lib/security/cacerts
/Library/Java/JavaVirtualMachines/jdk-9.0.4.jdk/Contents/Home/lib/security/cacerts: Java KeyStore
```

However, since the JDK wants to default to PKCS12 keystores, the JDK's [DualFormatPKCS12](https://github.com/AdoptOpenJDK/openjdk-jdk9/blob/master/jdk/src/java.base/share/classes/sun/security/pkcs12/PKCS12KeyStore.java#L137) will fall back to a JKS if reading a file as PKCS12 fails. Bouncycastle assumes that when the javax.net.ssl.trustStoreType is pkcs12 that is really what we mean.

Thus, to workaround this problem, set the `javax.net.ssl.trustStoreType` system property to `JKS`:

`-Djavax.net.ssl.trustStoreType=JKS`

However, it would be nice not to have to use a workaround like this. I filed an issue with bouncycastle as well: https://github.com/bcgit/bc-java/issues/663

**Comment by hborders** [comment]
[Bouncycastle doesn't want to make their PKCS12 spi read JKS files as Oracle did](https://github.com/bcgit/bc-java/issues/663#issuecomment-593230397), so Robolectric should try to handle this out of the box. Is it possible to manually set the set the `javax.net.ssl.trustStoreType` system property to `JKS`, or at least call this out in the setup instructions?

**Comment by hoisie** [comment]
This has been fixed in 4.4.
